paasta_tools/cli/cmds/mark_for_deployment.py

```diff
@@ -323,15 +366,57 @@ def instances_deployed(cluster, service, instances, git_sha):
                              .format(service, instance, cluster,
                                      status.marathon.running_instance_count,
                                      status.marathon.expected_instance_count))
+                instances_in.task_done()
+                instances_out.put(instance)
                 continue
             paasta_print("Complete: {}.{} on {} looks 100% deployed at {} "
                          "instances on {}"
                          .format(service, instance, cluster,
                                  status.marathon.running_instance_count,
                                  status.git_sha))
-            result.append(instance)
+            instances_in.task_done()
+
 
-    return result
+def _query_clusters(cluster_map, service, git_sha, green_light):
+    """Run _run_cluster_worker() in a separate thread for each paasta cluster
+
+    Args:
+        green_light: an instance of threading.Event(). It is used to terminate
+            all running threads after KeyboardInterrupt.
+    """
+    for cluster, cluster_data in cluster_map.items():
+        if cluster_data['deployed'] != cluster_data['num_instances']:
+            worker = Thread(target=_run_cluster_worker,
+                            args=(cluster, cluster_data, service, git_sha,
+                                  green_light))
+            worker.start()
+            cluster_data['worker'] = worker
```

Comment:
I can't think of any reason to use NamedTuples here.
Could you explain this in more details please?

cluster_map[cluster]['worker'] as well as cluster_map[cluster]['queue'] is a reference.
I'm using them in an internal function that is not supposed to call by anyone outside of wait-for-deployment.
What is wrong with keeping them in the same dict as other attributes?